scaaml/capture/input_generators/attack_point_iterator_models.py

```diff
@@ -114,3 +115,55 @@ def get_generated_keys(self) -> List[str]:
 
 BasicIteratorModels: TypeAlias = Union[ConstantIteratorModel,
                                        GeneratedIteratorModel]
+
+
+class RepeatIteratorModel(BaseModel):
+    """
+    Initialize the repeated iterate. If repetitions is not present
+    or set to a negative number it will do an infinite loop and
+    if it is 0 it will not repeat at all.
+          
+        Args:
+            operation (Literal['repeat']): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file.
+
+            configuration (Dict): The config for the iterated object
+                that will get repeated.
+                
+            repetitions (int): This parameter decides how many times the
+                iterator gets repeated. If it is a negative number it
+                will repeat infinitely and if you call __len__ it will
+                raise an LengthIsInfiniteException. If it is 0 then it will not
+                repeat at all. If it is a positive number it will
+                repeat that many times.
+    """
+    operation: Literal["repeat"]
+    repetitions: int
```

Comment:
Why are we changing the semantics here? `AttackPointIteratorRepeat.__init__` has default `repetitions: int = -1`.